onnxruntime/core/providers/openvino/backends/basic_backend.cc

```diff
@@ -96,6 +97,7 @@ BasicBackend::BasicBackend(std::unique_ptr<ONNX_NAMESPACE::ModelProto>& model_pr
     } else if (!session_context_.has_external_weights &&
                !subgraph_context_.has_dynamic_input_shape &&
                !session_context_.so_context_enable &&
+               session_context.shape.empty() &&
```

Comment:
As I see the model still has dynamic shape so this should not be here.

Reply:
And by keeping this here we are saying that we wont use unified compile model API for upper bound and lower bound models... which will impact FIL.